src/dataframe/dataframe.jl

```diff
@@ -698,36 +806,48 @@ end
 ##############################################################################
 
 # hcat! for 2 arguments, only a vector or a data frame is allowed
-function hcat!(df1::DataFrame, df2::AbstractDataFrame)
-    u = add_names(index(df1), index(df2))
+# TODO: after deprecation period change all to makeunique::Bool=false
+function hcat!(df1::DataFrame, df2::AbstractDataFrame; makeunique::Bool=true)
```

Comment:
Shouldn't we set `makeunique=false` immediately? Else, now deprecation warning will ever be printed, right?

Reply:
Now I get what you are getting at :).
I will set `makeunique` to `false` everywhere but it will change the code so that it has no consequences for now except printing deprecation. Right?

Reply:
Right!